### PR TITLE
published and unpublished

### PR DESCRIPTION
**Pull request by BenjaminCall**
# Code-review checklist #

  * Documentation
      * [ ] Interfaces describe how other programmers would use them.
      * [ ] Code that is not obvious has documentation describing why
        that code is the way it is.

### CODE REVIEW

File: client/js/_author/components/assessments/_edit_assessment.jsx [hunk 1]
```diff
@@ -112,10 +99,32 @@ export class NewAssessment extends React.Component {
     return { ...this.props.assessment, ...this.state.assessment };
   }
 
+  assignedAssessment(published) {
+    const { assessment } = this.props;
+    const editBankId = 'assessment.Bank%3A588f9225c89cd977c3560780%40ODL.MIT.EDU';
```
Comment (BenHSmith13) [inline client/js/_author/components/assessments/_edit_assessment.jsx]: We should extract these to somewhere

File: client/js/_author/components/assessments/_edit_assessment.jsx [hunk 2]
```diff
@@ -112,10 +99,32 @@ export class NewAssessment extends React.Component {
...
+      this.props.deleteAssignedAssessment(assessment, editBankId);
+      // Need to delete the editBankId and then add the publishedBankId
+      this.props.assignedAssessment(assessment, publishedBankId);
+    }
+  }
+
   render() {
+    // debugger
+    const publishedBankId = 'assessment.Bank%3A588f9240c89cd977c3560781%40ODL.MIT.EDU';
+    const isPublished = _.findIndex(this.props.currentAssessment.assignedBankIds, (id) => { return id === publishedBankId; }) !== -1 ? true : false;
```
Comment (BenHSmith13) [inline client/js/_author/components/assessments/_edit_assessment.jsx]: you can just use the boolean value of this, without the turnary

File: client/js/_author/components/bank_navigation/bank_list_item.jsx [hunk 3]
```diff
@@ -5,6 +5,9 @@ import Icon       from './bank_icon';
 // TODO: think about breaking this into smaller components
 export default function bankListItem(props) {
   const { bank } = props;
+  // const editBankId = 'assessment.Bank%3A588f9225c89cd977c3560780%40ODL.MIT.EDU';
+  const publishedBankId = 'assessment.Bank%3A588f9240c89cd977c3560781%40ODL.MIT.EDU';
+  const isPublished = _.findIndex(bank.assignedBankIds, (id) => { return id === publishedBankId; }) !== -1 ? true : false;
```
Comment (BenHSmith13) [inline client/js/_author/components/bank_navigation/bank_list_item.jsx]: same, extract, and unnecessary turnary

File: client/js/_author/components/assessments/_edit_assessment.jsx [hunk 4]
```diff
@@ -99,10 +114,31 @@ export class NewAssessment extends React.Component {
     return { ...this.props.assessment, ...this.state.assessment };
   }
 
+  assignedAssessment(published) {
```
Comment (dittonjs) [inline client/js/_author/components/assessments/_edit_assessment.jsx]: i think that a more discriptive name is needed for this. The name doesnt really describe what this method does. like `editOrPublishAssignedAssessment`

**Comment by dittonjs** [comment]
rename that stuff and looks good

File: client/js/_author/components/bank_navigation/_bank_navigator.jsx [hunk 5]
```diff
@@ -89,6 +94,7 @@ export class BankNavigator extends React.Component {
   }
 
   render() {
+    debugger
```
Comment (dittonjs) [inline client/js/_author/components/bank_navigation/_bank_navigator.jsx]: debugger

File: client/js/_author/components/assessments/_new_assessment.jsx [hunk 6]
```diff
@@ -6,22 +6,28 @@ import * as BankActions       from '../../../actions/qbank/banks';
 import * as AssessmentActions from '../../../actions/qbank/assessments';
 import * as ItemActions       from '../../../actions/qbank/items';
 
-function select() {
+function select(state) {
   return {
-
+    settings: state.settings,
```
Comment (BenHSmith13) [inline client/js/_author/components/assessments/_new_assessment.jsx]: why not just pull out editableBankId?

**Pull request merged**